### DISCUSSION TITLE
Possible to link to line number?

### DISCUSSION THREAD

**Discussion by dstengle** [foambubble/foam#1335] [Ideas]
As part of adding dynamic queries, it makes sense to be able to link directly to individual task items. Is this already possible some way in wikilinks? Standard []() linking supports it but not in the preview and wikilinks don't seem to work.

If not currently possible, is it much of a lift to add? It seems like it should be possible given the heading linking.

**Comment by riccardoferretti** [comment]
I am not aware of a way of doing that in markdown. 
In github links to line numbers are encoded as `#L<n>`, e.g. `#L2` for line 2. 
In Foam this could clash with a section called `L2` (in github I don't think it would be a problem, as section names are slugified, which includes going lowercase - but I haven't tested it) - see [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet).

Despite the conflict, I can still see how that would be useful, with the priority being given to a section called `L<n>` if present, and the line used as backup. Yes, there is a small set of cases that might not work well, but it might be a decent tradeoff, if there is value in having a standard to reference lines.

Anoption would be to use something along the lines, or compatible, with block anchors (e.g. `[[Test#^3asf]]`). Block level linking has been spoken about a lot in Foam, but never implemented. A line could be something like `[[Test#^L3]]`, not compatible with gh, but conflict free.

All in all, it depends on the specific goal.

Separately, although I understand how linking to lines can help with tasks, I wonder if it might be a bit brittle? What's your thinking around that?

**Reply by dokmanc** [reply]
If the markdown preview supported protocols like file:// or vscode:// , for example, you could do something like this to jump to line 3.
vscode://file/c:/Users/YourUsername/projects/MyProject/file.cs#L3
Markdown Preview Enhanced extension has the ability to specify in settings what protocols are supported.  For example: 
Markdown-preview-enhanced: Protocols White List
I entered: http://, https://, atom://, file://, mailto:, tel:, chad://
While most people would use the file:// and vscode:// protocols, I created my own called "chad" to not just jump to a specific line number but to jump to a code document line that contains a GUID comment. If the line number shifts, the hyperlink still takes you to that line. That way, I can document my code with external markdown files.

I would use Markdown Enhanced Preview Extension if it supported Wiki Links and all the nice features that come with FOAM. I'm really trying to create an Obsidian like environment in a company where Obsidian is not allowed and they thing I should use OneNote instead because it "does the same thing: make a note"
